File: basic/bruteforce/virus/virus.py
import sys

def dfs(graph, visited, node):
    visited[node] = True
    count = 0
    for neighbor in graph[node]:
        if not visited[neighbor]:
            count += 1 + dfs(graph, visited, neighbor)
    return count

def solve():
    n = int(sys.stdin.readline())
    m = int(sys.stdin.readline())
    graph = [[] for _ in range(n + 1)]
    for _ in range(m):
        u, v = map(int, sys.stdin.readline().split())
        graph[u].append(v)
        graph[v].append(u)

    visited = [False] * (n + 1)
    infected_count = dfs(graph, visited, 1)
    print(infected_count)

if __name__ == "__main__":
    solve()

File: basic/bruteforce/virus/virus.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const n = parseInt(input[0]);
const m = parseInt(input[1]);
const graph = Array.from({ length: n + 1 }, () => []);
const visited = new Array(n + 1).fill(false);
let count = 0;

for (let i = 2; i < 2 + m; i++) {
    const [u, v] = input[i].split(" ").map(Number);
    graph[u].push(v);
    graph[v].push(u);
}

function dfs(node) {
    visited[node] = true;
    for (const neighbor of graph[node]) {
        if (!visited[neighbor]) {
            count++;
            dfs(neighbor);
        }
    }
}

dfs(1);
console.log(count);